servers/navigation_2d_server.cpp

```diff
@@ -0,0 +1,237 @@
+		PoolVector<Vector3> verts;
+		{
+			verts.resize(d->get_vertices().size());
+			PoolVector<Vector3>::Write w = verts.write();
+
+			PoolVector<Vector2>::Read r = d->get_vertices().read();
+
+			for (int i(0); i < d->get_vertices().size(); i++) {
+				w[i] = v2_to_v3(r[i]);
+			}
+		}
+		ret->set_vertices(verts);
+
+		{
+			for (int i(0); i < d->get_polygon_count(); i++) {
+				ret->add_polygon(d->get_polygon(i));
+			}
+		}
+	}
+	return ret;
+}
+
+void Navigation2DServer::_bind_methods() {
+	ClassDB::bind_method(D_METHOD("map_create"), &Navigation2DServer::map_create);
+	ClassDB::bind_method(D_METHOD("map_set_active", "map", "active"), &Navigation2DServer::map_set_active);
+	ClassDB::bind_method(D_METHOD("map_is_active", "nap"), &Navigation2DServer::map_is_active);
+	ClassDB::bind_method(D_METHOD("map_set_cell_size", "map", "cell_size"), &Navigation2DServer::map_set_cell_size);
+	ClassDB::bind_method(D_METHOD("map_get_cell_size", "map"), &Navigation2DServer::map_get_cell_size);
+	ClassDB::bind_method(D_METHOD("map_set_edge_connection_margin", "map", "margin"), &Navigation2DServer::map_set_edge_connection_margin);
+	ClassDB::bind_method(D_METHOD("map_get_edge_connection_margin", "map"), &Navigation2DServer::map_get_edge_connection_margin);
+	ClassDB::bind_method(D_METHOD("map_get_path", "map", "origin", "destination", "optimize"), &Navigation2DServer::map_get_path);
+
+	ClassDB::bind_method(D_METHOD("region_create"), &Navigation2DServer::region_create);
+	ClassDB::bind_method(D_METHOD("region_set_map", "region", "map"), &Navigation2DServer::region_set_map);
+	ClassDB::bind_method(D_METHOD("region_set_transform", "region", "transform"), &Navigation2DServer::region_set_transform);
+	//ClassDB::bind_method(D_METHOD("region_set_navmesh", "region", "nav_mesh"), &Navigation2DServer::region_set_navmesh);
+	//ClassDB::bind_method(D_METHOD("region_bake_navmesh", "mesh", "node"), &Navigation2DServer::region_bake_navmesh);
+
+	ClassDB::bind_method(D_METHOD("agent_create"), &Navigation2DServer::agent_create);
+	ClassDB::bind_method(D_METHOD("agent_set_map", "agent", "map"), &Navigation2DServer::agent_set_map);
+	ClassDB::bind_method(D_METHOD("agent_set_neighbor_dist", "agent", "dist"), &Navigation2DServer::agent_set_neighbor_dist);
+	ClassDB::bind_method(D_METHOD("agent_set_max_neighbors", "agent", "count"), &Navigation2DServer::agent_set_max_neighbors);
+	ClassDB::bind_method(D_METHOD("agent_set_time_horizon", "agent", "time"), &Navigation2DServer::agent_set_time_horizon);
+	ClassDB::bind_method(D_METHOD("agent_set_radius", "agent", "radius"), &Navigation2DServer::agent_set_radius);
+	ClassDB::bind_method(D_METHOD("agent_set_max_speed", "agent", "max_speed"), &Navigation2DServer::agent_set_max_speed);
+	ClassDB::bind_method(D_METHOD("agent_set_velocity", "agent", "velocity"), &Navigation2DServer::agent_set_velocity);
+	ClassDB::bind_method(D_METHOD("agent_set_velocity_target", "agent", "target_velocity"), &Navigation2DServer::agent_set_target_velocity);
```

Comment:
I think it should be either `target_velocity` or `velocity_target` for consistency. Same for the 3D variant. I like the latter more because it'd show up next to `agent_set_velocity` in the docs (although I'm not sure there is any naming conventions in use.)

Reply:
I think "velocity target" is the least confusing. "Target velocity" sounds similar to "target's velocity" (velocity of some other guy besides the agent), while "velocity target" is more obviously the level of velocity to be reached.

Reply:
Also, shouldn't it be `set_agent_radius` instead of `agent_set_radius`? I think there's a discussion about this specifically somewhere, and I can't find it, but https://github.com/godotengine/godot/issues/33026 is related.

Reply:
When I see such naming I think of the convention used in servers in general, as they are close to a C-like API, so because they use to take a RID as "self", the type of "self" would be prepended as the first word of the function's name